Change postgresql.useSubChart to .enabled

mitchnielsen opened the pull request:
## Summary

Changes `postgresql.useSubChart` to `postgresql.enabled`, following the common convention mentioned in https://helm.sh/docs/topics/charts/https://helm.sh/docs/topics/charts/.

This should hopefully make it clearer what that key does - and this key already existed, so by removing `useSubChart`, there's only one key to manipulate.

Other commits in this PR have commit messages explaining what and why.

Closes https://github.com/PrefectHQ/prefect-helm/issues/336

## Testing

### Internal PostgreSQL enabled (default)

(Default values used)

```
$ helm template test . -f test.values.yaml --show-only charts/postgresql/templates/primary/statefulset.yaml | yq '.metadata.name'
test-postgresql
```

### Internal PostgreSQL disabled

```yaml
postgresql:
  enabled: false
```

```
$ helm template test . -f test.values.yaml --show-only charts/postgresql/templates/primary/statefulset.yaml | yq '.metadata.name'
Error: could not find template charts/postgresql/templates/primary/statefulset.yaml in chart
null
```

### Internal PostgreSQL + default Secret

(Default values used)

```
$ helm template test . -f test.values.yaml --show-only templates/deployment.yaml| yq '.spec.template.spec.containers[0].env[] | select(.name == "PREFECT_API_DATABASE_CONNECTION_URL") | .valueFrom.secretKeyRef.name'
prefect-server-postgresql-connection
```

### Internal PostgreSQL + existing Secret

```yaml
postgresql:
  enabled: true
  auth:
    existingSecret: my-external-pg-secret

```

```
$ helm template test . -f test.values.yaml --show-only templates/deployment.yaml| yq '.spec.template.spec.containers[0].env[] | select(.name == "PREFECT_API_DATABASE_CONNECTION_URL") | .valueFrom.secretKeyRef.name'
my-external-pg-secret
```

### External PostgreSQL + existing Secret

```yaml
postgresql:
  enabled: false
  auth:
    existingSecret: my-external-pg-secret
```

```
$ helm template test . -f test.values.yaml --show-only templates/deployment.yaml| yq '.spec.template.spec.containers[0].env[] | select(.name == "PREFECT_API_DATABASE_CONNECTION_URL") | .valueFrom.secretKeyRef.name'
my-external-pg-secret
```

mitchnielsen committed:
Change postgresql.useSubChart to .enabled

Changes 'useSubChart' to 'enabled'https://helm.sh/docs/topics/charts/
to follow common convention as mentioned in
https://helm.sh/docs/topics/charts/.

This should make it more clear what this key does.

Closes https://github.com/PrefectHQ/prefect-helm/issues/336

mitchnielsen committed:
Remove postgresql.enabled check on Secret injection

Removes the postgresql.enabled check on the PostgreSQL Secret injection
in the Deployment.

Turning off the installation of the chart shouldn't prevent a secret
from being injected, since a user can provide an existing Secret name.

mitchnielsen commented:
Jotting down action items for myself:

Caught up with @jamiezieziula on this. The `postgresql.auth` key in `values.yaml` is Prefect-specific, so we need to allow `auth` settings to be set outside of the scope of `postgresql` when the built-in PG chart isn't used (`postgresql.enabled=false`).

We'll also want to support passing no auth and no database for those who want to test with the SQLite DB.

mitchnielsen committed:
Document how to use external PostgreSQL instance

Documents how to disable the bundled PostgreSQL instance and provide the
auth Secret.

This Secret will be injected into the Prefect Server Deployment.

mitchnielsen committed:
Docs: remove 'password' key from existingSecret

Looking at the Secret template, we no longer need a `password` key in
the existingSecret object.

jamiezieziula commented on charts/prefect-server/values.schema.json:
```diff
@@ -660,9 +660,9 @@
           "form": true
         },
-        "useSubChart": {
+        "enabled": {
```
this already exists on line 599

mitchnielsen replied:
Good catch, removed in 351e405.

jamiezieziula reviewed (changes requested):
There is a duplicate `enabled` map in the json schema. Also, can you remove all `form: true` fields in the schema, these aren't needed (we removed most before but must have missed these)

mitchnielsen committed:
Remove extra 'enabled' key from schema

Removes the extra 'enabled' key from the schema as it's already defined
on line 599.

mitchnielsen committed:
Remove 'form' fields from the server schema

mitchnielsen commented:
Removed `form` fields from the server schema in db29dcf.

mitchnielsen commented:
- Failures seem to be from hitting the Docker pull limit.
- Will proceed with merge soon but wanted to give the community member a chance to chime in.
- Will also consider if this could be a breaking change 🤔

pythonking6 commented:
This is excellent!!!
